packages/blade/src/components/DatePicker/DateInput.web.tsx

```diff
@@ -0,0 +1,184 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useDatesContext, getFormattedDate } from '@mantine/dates';
+import React from 'react';
+import BaseBox from '~components/Box/BaseBox';
+import type { FormInputValidationProps } from '~components/Form';
+import { ArrowRightIcon, CalendarIcon } from '~components/Icons';
+import type { BaseInputProps } from '~components/Input/BaseInput';
+import { BaseInput } from '~components/Input/BaseInput';
+import { size } from '~tokens/global';
+import { makeSize } from '~utils';
+import type { BladeElementRef } from '~utils/types';
+
+const _DateInput = (
+  props: BaseInputProps,
+  ref: React.ForwardedRef<BladeElementRef>,
+): React.ReactElement => {
+  return (
+    <BaseInput
+      {...props}
+      ref={ref}
+      as="button"
+      textAlign="left"
+      hideLabelText={props.label?.length === 0}
+      autoCompleteSuggestionType="none"
+      hasPopup="dialog"
+      onClick={(e) => {
+        if (props.isDisabled) {
+          return;
+        }
+        props.onClick?.(e);
+      }}
+      onKeyDown={({ event }) => {
+        // @ts-expect-error
+        props.onKeyDown?.(event);
+      }}
+    />
+  );
+};
+
+const DateInput = React.forwardRef(_DateInput);
+
+type DatePickerRangeInputProps = {
+  selectionType: 'range';
+  label?: { start: string; end?: string };
+  date: [Date, Date];
+};
+type DatePickerSingleInputProps = {
+  selectionType: 'single';
+  label?: string;
+  date: Date;
+};
+
+type DatePickerCommonInputProps = {
+  labelPosition?: BaseInputProps['labelPosition'];
+  inputRef: React.Ref<any>;
+  referenceProps: any;
+} & Pick<BaseInputProps, 'size' | 'isRequired' | 'isDisabled' | 'accessibilityLabel'> &
+  FormInputValidationProps;
+
+type DatePickerInputProps = DatePickerCommonInputProps &
+  (DatePickerRangeInputProps | DatePickerSingleInputProps);
+
+const iconVerticalMargin = {
+  medium: size[16],
+  large: size[24],
+} as const;
+const _DatePickerInput = (
+  {
+    selectionType,
+    referenceProps,
+    inputRef,
+    date,
+    label,
+    labelPosition,
+    size = 'medium',
+    ...props
+  }: DatePickerInputProps,
+  ref: React.ForwardedRef<any>,
+): React.ReactElement => {
+  const format = 'DD/MM/YYYY';
```

Comment:
format and placeholder don't seem to match

Reply:
fixed